src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeModal.js

```diff
@@ -291,11 +291,22 @@ export const GcpComputeModalBase = ({
   )(async () => {
     const { runtime: existingRuntime, persistentDisk: existingPersistentDisk } = getExistingEnvironmentConfig();
     const { runtime: desiredRuntime, persistentDisk: desiredPersistentDisk } = getDesiredEnvironmentConfig();
+
+    // Whether applyChanges will send a PATCH request to the disk. Will only happen if runtime is updated or created.
     const shouldUpdatePersistentDisk = canUpdatePersistentDisk() && !_.isEqual(desiredPersistentDisk, existingPersistentDisk);
-    const shouldDeletePersistentDisk = existingPersistentDisk && !canUpdatePersistentDisk();
-    const shouldUpdateRuntime = canUpdateRuntime() && !_.isEqual(desiredRuntime, existingRuntime);
-    const shouldDeleteRuntime = existingRuntime && !canUpdateRuntime();
-    const shouldCreateRuntime = !canUpdateRuntime() && !!desiredRuntime;
+
+    // Whether applyChanges will send a DELETE request to the disk. Can happen alone or be followed by a runtime create.
+    const shouldDeletePersistentDisk = !canUpdatePersistentDisk() && !!existingPersistentDisk;
+
+    // Whether applyChanges will send a PATCH request to the runtime.
+    const shouldUpdateRuntime = canUpdateRuntime();
```

Comment:
this is the fix: removed the equality check. if only disk size changes, we should still update the runtime.